extensions/exposition/documentation/io.md

```diff
@@ -0,0 +1,56 @@
+# I/O restrictions
+
+The Exposition comes with `io` directives to control access to the operation's input and output
+properties.
+
+## `io:input`
+
+The `io:input` optional directive contains a list of properties that are allowed to be specified in
+the request body.
+
+```yaml
+POST:
+  endpoint: create
+  io:input: [name, location]
+```
+
+The list must be a valid subset of the operation's input properties.
+
+If `io:input` is specified and the request body is not an object, or contains properties that are
+not in the list, the request will be rejected with a `400` status code.
```

Comment:
This is VERY confusion that Toa is using JSON schemas everywhere to enforce that data shape but here we want some sort of self invented subset of features. In most cases such restrictions can be expressed using JSON schema dependencies such as if-then-else.

This will also make impossible to generate proper TS types of Toa manifest if that feature will be landed.

Reply:
This is not a subset of JSON schema, this is basically an array of string. It has nothing to do with data types validation.

What's good in this filter, unlike JSON schema here, is that it **can** be applied to data **and** to its schema to make TS types generation possible.

Also, AFAIR, shrinking the data has nothing in common with JSON Schema itself, it's just custom function of specific validator implementations.